code/datums/elements/climbable.dm

```diff
@@ -82,8 +84,18 @@
 
 
 /datum/element/climbable/proc/do_climb(atom/climbed_thing, mob/living/user)
+	if(!can_climb(climbed_thing, user))
+		return
 	climbed_thing.set_density(FALSE)
-	. = step(user, get_dir(user,climbed_thing.loc))
+	var/dir_step = get_dir(user, climbed_thing.loc)
+	//it's on border you are actually vaulting over it to the direction it's facing.
+	if(climbed_thing.flags_1 & ON_BORDER_1 && (climbed_thing.loc == user.loc || !(dir_step & REVERSE_DIR(climbed_thing.dir))))
+		//can be vaulted over in two different cardinal directions.
+		if((climbed_thing.dir in GLOB.diagonals) && climbed_thing.loc == user.loc)
+			dir_step = (user.dir & climbed_thing.dir) || angle2dir_cardinal(dir2angle(climbed_thing.dir))
```

Comment:
What are you trying to do here with `angle2dir_cardinal(dir2angle(climbed_thing.dir))`? This is just converting to angle a diagonal value, and then prioritizing a pre-defined cardinal side.
It's pretty obscure.

Normally what you want to do is test both cardinal directions, for diagonal moves.